### PR TITLE
add selection str parser

### PR DESCRIPTION
**Pull request by kain88-de**
This adds a parser for tag selections to use

b.tags['apple and ham or tuna']

@dotsdl I hope this gives you some motivation to finish up the spring cleaning.

### CODE REVIEW

File: src/datreant/core/limbs.py [hunk 1]
```diff
@@ -87,17 +88,25 @@ def _setter(self, val):
...
+        return self._getselection(value)
+
+    def _getselection(self, value):
+        """get tags according to a selection
+        """
         with self._read:
             if isinstance(value, list):
                 # a list of tags gives only members with ALL the tags
-                fits = all([self[item] for item in value])
+                fits = all([self._getselection(item) for item in value])
```
Comment (richardjgowers) [inline src/datreant/core/limbs.py]: With the `any` and `all` calls, you can pass in a generator (ie remove the square brackets) so they can exit early

**Comment by kain88-de** [comment]
OK I now got a better working version. The [gist](https://gist.github.com/kain88-de/9728e5b76082d9c3336ce840cb3195b5) is also updated

**Comment by kain88-de** [comment]
@dotsdl I fixed tag parser now.  It works for my own datreants to give correct results and I find the tag selection much more understandable now.

**Comment by dotsdl** [comment]
@kain88-de looks good to me. Thanks for putting this together!

**Comment by dotsdl** [comment]
@kain88-de even though it's already merged, how does this deal with tags that have spaces? I know that seems like a stupid choice for a tag, but thinking of failure modes here. Also, what about tags that have spaces and words like 'or'? Should we not even bother supporting tags like that, and give an error if they're attempted?

One option is to not even allow spaces in tags in the first place on addition. We currently only guarantee that they are strings.

**Comment by kain88-de** [comment]
Currently the parser code will return the original string if parsing fails. This alllows stuff like 'foo bar' to still work. But the problem is we do that silently for all parser errors.  Maybe a warning would be good. 

  But anything you enclose in quotation marks will be read as a single word bei the parser so `foo and 'bar bar'` should work fine.  I haven't checked yet if this also allows the DSL keywords in quotes for names.

**Comment by kain88-de** [comment]
Yes anything that you enclose in quotation marks is read as a single word so `'"food and drinks"'` is parsed to `food and drinks` while `'food and drinks'` is parsed to `['food', 'drinks']`

**Comment by dotsdl** [comment]
Ah cool, works fine then! That's satisfying.

**Pull request merged**